jep-012-01-raw-string-literals.md

```diff
@@ -0,0 +1,358 @@
+# Raw String Literals
+
+|||
+|---|---
+| **JEP**    | 12-01
+| **Author** | Michael Downling, Maxime Labelle
+| **Status** | accepted
+| **Created**| 19-Nov-2022
+| **Obsoletes**| [JEP-12](./jep-012-raw-string-literals.md)
+
+## Abstract
+
+This JEP proposes the following modifications to JMESPath in order to improve
+the usability of the language and ease the implementation of parsers:
+
+
+* Addition of a **raw string literal** to JMESPath that will allow expressions
+to contain raw strings that are not mutated by JSON escape sequences (e.g.,
+“\\n”, “\\r”, “\\u005C”).
+
+
+* Deprecation of  the current literal parsing behavior that allows for unquoted
+JSON strings to be parsed as JSON strings, removing an ambiguity in the
+JMESPath grammar and helping to ensure consistency among implementations.
+
+This proposal seeks to add the following syntax to JMESPath:
+
+```
+'foobar'
+'foo\'bar'
+`bar` -> Parse error/warning (implementation specific)
+```
+
+## Motivation
+
+Raw string literals are provided in [various programming languages](https://en.wikipedia.org/wiki/String_literal#Raw_strings) in order to prevent
+language specific interpretation (i.e., JSON parsing) and remove the need for
+escaping, avoiding a common problem called [leaning toothpick syndrome (LTS)](https://en.wikipedia.org/wiki/Leaning_toothpick_syndrome). Leaning toothpick
+syndrome is an issue in which strings become unreadable due to excessive use of
+escape characters in order to avoid delimiter collision (e.g., `\\\\\\`).
+
+When evaluating a JMESPath expression, it is often necessary to utilize string
+literals that are not extracted from the data being evaluated, but rather
+statically part of the compiled JMESPath expression. String literals are useful
+in many areas, but most notably when invoking functions or building up
+multi-select lists and hashes.
+
+The following expression returns the number of characters found in the string
+`"foo"`. When parsing this expression, `` `"foo"` `` is parsed as a JSON value
+which produces the string literal value of `foo`:
+
+```
+`"foo"`
+```
+
+The following expression is functionally equivalent. Notice that the quotes are
+elided from the JSON literal:
+
+```
+`foo`
+```
```

Comment:
`````suggestion
### Obsolete alternative

_This section recapitulates content from the original version of JEP-12 that is no longer accurate of of this replacement._

The following expression is functionally equivalent. Notice that the quotes are
elided from the JSON literal:

```
`foo`
```
`````

Reply:
Please, can you rephrase the `of of` as I do not know what you had in mind.

Reply:
Done. It was intended to be "as of".

Reply:
Agreed. ✅